dht_net.go

```diff
@@ -208,12 +210,38 @@ func (dht *IpfsDHT) handleNewMessage(s network.Stream) bool {
 	}
 }
 
+type messageManager struct {
```

Comment:
Mostly because I'm not really a fan of adding in all the complexity of the stream management into the a simple wrapper for sending protocol messages, but it was the easiest way to do it for now.

It'd be nice to make that all optional in the future, but the stream management and stream parsing are bit mixed together at the moment in the form of the `messageManager` struct.

Reply:
To make this more clear I've now decoupled the `messageManager` from the `ProtocolMessenger` via an interface. This should give consumers of `ProtocolMessenger` more flexibility to do things like immediately close connections after sending messages if they're not interested in long-term connections.